src/driver-hidpp20.c

```diff
@@ -1546,8 +1546,9 @@ hidpp20drv_commit(struct ratbag_device *device)
 					if (resolution->is_active)
 						dpi_index = resolution->index;
 				}
-				hidpp20_onboard_profiles_set_current_dpi_index(drv_data->dev,
-									       dpi_index);
+				if (dpi_index != 0xff)
```

Comment:
Couldn't we do instead of this (in my opinion a little bit hacky) solution do simply:

```C
if (resolution->is_active)
	hidpp20_onboard_profiles_set_current_dpi_index(drv_data->dev,
	                                               resolution->index);
```

And even remove the variable.

Reply:
I think we had that for previously but for some reason removed it. I am not home right now, could you look at the git blame?

Reply:
True, commit is 563550c473ff29849560007eec6808f2995152c2. If I get it right the problem was that `dpi_index` was used before in the commit loop and that could be a problem depending on the order of the profiles somehow. However, `dpi_index` is not used before anymore, so I think it should be fine to remove it.

Reply:
Thanks for looking into it, I am going to change it when I get home.

Reply:
agree with @stephanlachnit, whatever the code was before the current flow allows it to be called on `is_active` which makes the whole thing less confusing, IMO.